core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -486,18 +487,24 @@ private StartupConsistencyChecker(Long fileId) {
             inconsistentUris.add(parentUri);
           }
           for (Inode childInode : ((InodeDirectory) parentInode).getChildren()) {
-            AlluxioURI childUri = parentUri.join(childInode.getName());
-            if (childInode.isDirectory()) {
-              dirsToCheck.add(childInode.getId());
-            } else {
-              childInode.lockRead();
-              try {
+            try {
+              childInode.lockReadAndCheckParent(parentInode);
+            } catch (InvalidPathException e) {
+              // This should be safe, continue.
+              LOG.debug("A file scheduled for consistency check was moved before the check.");
```

Comment:
would it make sense to log e.message() instead of assuming the root case of this exception?